subprojects/openvpn3/openvpn3.c

```diff
@@ -409,21 +409,49 @@ void set_receive_log_events(char *session_object, int set_to)
 
 void set_log_forward(){
 
+    g_assert(UniqueSession != NULL);
+
     GError *error = NULL;
-    GVariant *result = g_dbus_proxy_call_sync(UniqueSession,
-                                              "net.openvpn.v3.sessions.LogForward",
-                                              g_variant_new("(b)", true),
-                                              G_DBUS_PROXY_FLAGS_NONE,
-                                              -1,
-                                              NULL,
-                                              NULL);
+    g_dbus_proxy_call_sync( UniqueSession,
+                            "net.openvpn.v3.sessions.LogForward",
+                            g_variant_new("(b)", true),
+                            G_DBUS_PROXY_FLAGS_NONE,
+                            -1,
+                            NULL,
+                            &error );
     
     if (error != NULL)
     {
         g_warning("%s:%d -> %s", __FUNCTION__, __LINE__, error->message);
         g_error_free(error);
         return;
     }
+
+}
+
+char* is_ready_to_connect(){
+
+    g_assert(UniqueSession != NULL);
+
+    GError *error = NULL;
+    g_dbus_proxy_call_sync( UniqueSession,
+                            "net.openvpn.v3.sessions.Ready",
+                            NULL,
+                            G_DBUS_PROXY_FLAGS_NONE,
+                            -1,
+                            NULL,
+                            &error );
+    
+    if (error != NULL)
+    {
+        g_warning("%s:%d -> %s", __FUNCTION__, __LINE__, error->message);
+        char* error_msg = error->message;
+        g_error_free(error);
+        return error_msg;
```

Comment:
This will probably fail.  GError is a pointer, which has a struct member pointing at a buffer which you free with the `g_error_free()` call.  And then you will return a pointer to an inaccessible memory address.

You probably need to use strdup(), and the caller of `is_ready_to_connect()` need to `free()` the return value to avoid a memory leak.